src/mongo/delegates/Updates.class.php

```diff
@@ -1204,6 +1233,14 @@ public function setTransactionLog(TransactionLog $transactionLog)
         $this->transactionLog = $transactionLog;
     }
 
+    /**
+     * Register save changes event hooks
+     * @param IEventHook $hook
+     */
+    public function registerSaveChangesEventHook(IEventHook $hook)
+    {
+        $this->saveChangesHooks[] = $hook;
+    }
```

Comment:
How about having a generic register hook method to store the hooks in a multidimensional array, that way we won't have to add/manage different register methods for each new hook.